aszczachor/src/components/modal/Modal.js

```diff
@@ -0,0 +1,63 @@
+import styles from "components/modal/modal.module.css";
+import prevIcon from "assets/modal/prev-icon.png";
+import nextIcon from "assets/modal/next-icon.png";
+import closeIcon from "assets/modal/close-icon.png";
+import { useEffect, useRef } from "react";
+
+const Modal = ({ selected, setSelected, photos, setOpenModal }) => {
+  const nextPhoto = () => {
+    setSelected(selected === photos.length - 1 ? 0 : Number(selected) + 1);
+  };
+
+  const prevPhoto = () => {
+    setSelected(selected === 0 ? photos.length - 1 : Number(selected) - 1);
+  };
+
+  const closeModal = () => {
+    setOpenModal(false);
+    setSelected(null);
+  };
+
+  const left = useRef();
+  const right = useRef();
+  const image = useRef();
+
+  useEffect(() => {
+    const onModalClick = ({ target }) => {
+      if (
+        (image.current && image.current.contains(target)) ||
+        (left.current && left.current.contains(target)) ||
+        (right.current && right.current.contains(target))
+      ) {
+        return;
+      }
+      closeModal();
+    };
+    document.addEventListener("click", onModalClick, true);
+    return () => document.removeEventListener("click", onModalClick, true);
+  });
```

Comment:
brak tablicy dependencji w useEffect? chcemy to rerenderować za każdą zmianą? Tak ma być?

Reply:
tutaj dobrze by było renderować przy zmianie zmiennej isModalOpen?

Reply:
w sumie dla modala to byłoby git samo [], tylko przy pierwszym renderze, tylko coś krzyczy wtedy o missing dependencies musze jeszcze to ogarnac